e2e/cypress/integration/messaging/message_auto_response_spec.js

```diff
@@ -0,0 +1,109 @@
+        cy.apiUpdateConfig({TeamSettings: {ExperimentalEnableAutomaticReplies: true}});
+
+        // # Create a new team
+        cy.apiInitSetup().then(({team, user}) => {
+            userA = user;
+            testTeam = team;
+
+            // # Create a second user
+            cy.apiCreateUser().then(({user: createdUser}) => {
+                userB = createdUser;
+                cy.apiAddUserToTeam(team.id, userB.id);
+            });
+
+            offTopicChannelUrl = `/${testTeam.name}/channels/town-square`;
+        });
+    });
+
+    it('MM-T4004_1 Login as userB and set auto response message', () => {
+        // # Login as userB
+        cy.apiLogin(userB);
+
+        // # Visit off topic channel
+        cy.visit(offTopicChannelUrl);
+
+        // # Open 'Account Settings' modal and view 'Notifications'
+        cy.uiOpenAccountSettingsModal('Notifications').within(() => {
+            // # Click on 'Edit' for 'Automatic Direct Message Replies
+            cy.get('#auto-responderEdit').should('be.visible').click();
+
+            // # Click on 'Enabled' checkbox
+            cy.get('#autoResponderActive').should('be.visible').click();
+
+            // # Clear default auto response message
+            cy.get('#autoResponderMessageInput').should('be.visible').clear();
+
+            // # Enter new Auto Response Message
+            cy.get('#autoResponderMessageInput').should('be.visible').type(AUTO_RESPONSE_MESSAGE);
+
+            // # Save and close the modal
+            cy.uiSaveAndClose();
+        });
+
+        // # Logout userB
+        cy.apiLogout();
+    });
+
+    it('MM-T4004_1 Login as userA and send direct messages and verify only one auto response message is sent from userB', () => {
+        // # Login as userA
+        cy.apiLogin(userA);
+
+        // # Visit off topic channel
+        cy.visit(offTopicChannelUrl);
+
+        // # Send direct message to userB
+        cy.uiAddDirectMessage().click();
+        cy.get('#selectItems').type(`${userB.username}`);
+        cy.findByText('Loading').should('be.visible');
+        cy.findByText('Loading').should('not.exist');
+        cy.get('#multiSelectList').findByText(`@${userB.username}`).click();
+        cy.findByText('Go').click();
+        cy.postMessage(MESSAGE);
+
+        // * Verify if auto response message is last post displayed
+        cy.getLastPostId().then((replyId) => {
+            cy.get(`#postMessageText_${replyId}`).should('be.visible').and('have.text', AUTO_RESPONSE_MESSAGE);
+        });
+
+        // # Send another direct message to userB
+        cy.postMessage(MESSAGE);
+
+        // * Verify if auto response message is last post displayed
+        cy.getLastPostId().then((replyId) => {
+            cy.get(`#postMessageText_${replyId}`).should('be.visible').and('have.text', MESSAGE);
+        });
+
+        // # Send another direct message to userB
+        cy.postMessage(MESSAGE);
```

Comment:
`MESSAGE` should be good but would be better to post a different text to ensure we're verifying the real last post and not the second to the last text message.